src/jit/lclvars.cpp

```diff
@@ -3292,6 +3292,13 @@ void Compiler::lvaSortByRefCount()
         {
             varDsc->lvTracked = 0;
         }
+#if defined(JIT32_GCENCODER) && defined(WIN64EXCEPTIONS)
+        else if (lvaIsOriginalThisArg(lclNum))
+        {
+            // For x86/Linux, we cannot have this pointer in tracked variables, so we set this pointer untracked
+            varDsc->lvTracked = 0;
```

Comment:
Does it need to be untracked in all cases? What if there is no EH and/or no generics context? In that case, the VM doesn't need to look up `this` in the GC info, so we shouldn't need to treat is specially.